tool/tctl/common/plugins_command.go

```diff
@@ -96,11 +171,147 @@ func (p *PluginsCommand) Cleanup(ctx context.Context, clusterAPI *auth.Client) e
 	return nil
 }
 
+type samlConnectorsClient interface {
+	GetSAMLConnector(ctx context.Context, id string, withSecrets bool) (types.SAMLConnector, error)
+}
+
+type pluginsClient interface {
+	CreatePlugin(ctx context.Context, in *pluginsv1.CreatePluginRequest, opts ...grpc.CallOption) (*emptypb.Empty, error)
+}
+
+type installPluginArgs struct {
+	samlConnectors samlConnectorsClient
+	plugins        pluginsClient
+}
+
+func (p *PluginsCommand) InstallOkta(ctx context.Context, args installPluginArgs) error {
+	oktaSettings := p.install.okta
+
+	if oktaSettings.accessListSync {
+		if len(oktaSettings.defaultOwners) == 0 {
+			return trace.BadParameter("AccessList sync requires at least one default owner to be set")
+		}
+	}
+
+	if oktaSettings.scimToken != "" {
+		if len(oktaSettings.defaultOwners) == 0 {
+			return trace.BadParameter("SCIM support requires at least one default owner to be set")
+		}
+
+		if oktaSettings.appID == "" {
+			return trace.BadParameter("SCIM support requires app-id to be set")
+		}
+
+		if oktaSettings.samlConnector == "" {
+			return trace.BadParameter("SCIM support requires saml-connector to be set")
+		}
+	}
+
+	if oktaSettings.samlConnector != "" {
```

Comment:
if samlConnector was provided can we get the okta org from connector spec ?

Reply:
We can try it, and give a warning to set it explicitly if it can't be found. The App ID label is only guaranteed to be set if the SAML Connector was created though the Okta install flow; if it was created manually it may not exist.